Examples/Algorithms/TrackFinding/include/ActsExamples/TrackFinding/AdaptiveHoughTransformSeeder.hpp

```diff
@@ -0,0 +1,478 @@
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/EventData/SimSpacePoint.hpp"
+#include "ActsExamples/Framework/DataHandle.hpp"
+#include "ActsExamples/Framework/IAlgorithm.hpp"
+#include "ActsExamples/Framework/ProcessCode.hpp"
+
+#include <cstddef>
+#include <list>
+#include <memory>
+#include <numbers>
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace ActsExamples {
+struct AlgorithmContext;
+}
+
+namespace Acts {
+class TrackingGeometry;
+}
+
+namespace ActsExamples {
+// Helper class describing one section of the accumulator space
+class AccumulatorSection {
+ public:
+  AccumulatorSection() = default;
+
+  AccumulatorSection(float xw, float yw, float xBegin, float yBegin,
+                     int div = 0, const std::vector<unsigned> &indices = {},
+                     const std::vector<float> &history = {});
+
+  /// @brief keep indices and update parameters of the box
+  /// This method is useful when changing direction of the search
+  void updateDimensions(float xw, float yw, float xBegin, float yBegin);
+
+  /// @brief keep indices and update parameters of the box by scalling
+  /// @param xs - scale in x direction, if bigger than 1 the size increases
+  /// @param ys - scale in y direction
+  /// The box is recentred
+  void expand(float xs, float ys);
+
+  inline unsigned count() const {
+    return static_cast<unsigned>(m_indices.size());
+  }
+  inline const std::vector<unsigned> &indices() const { return m_indices; }
+  inline std::vector<unsigned> &indices() { return m_indices; }
+
+  // create section that is bottom part this this one
+  // +------+
+  // |      |
+  // +------+
+  // |     <|-- this part
+  // +------+
+  AccumulatorSection bottom(float yFraction = 0.5) const;
+  // see @bottom
+  AccumulatorSection top(float yFraction = 0.5) const;
+  // @see @bottom
+  AccumulatorSection left(float xFraction = 0.5) const;
+  // @see @bottom
+  AccumulatorSection right(float xFraction = 0.5) const;
+
+  // create section that is bottom left corner of this this one
+  // by default the section is divided into 4 quadrants,
+  // if parameters are provided the quadrants size can be adjusted
+  // +---+---+
+  // |   |   |
+  // +---+---+
+  // |   |  <|-- this part
+  // +---+---+
+  AccumulatorSection bottomRight(float xFraction = 0.5,
+                                 float yFraction = 0.5) const;
+  AccumulatorSection bottomLeft(float xFraction = 0.5,
+                                float yFraction = 0.5) const;
+
+  AccumulatorSection topLeft(float xFraction = 0.5,
+                             float yFraction = 0.5) const;
+  AccumulatorSection topRight(float xFraction = 0.5,
+                              float yFraction = 0.5) const;
+
+  /// @brief true if the line defined by given parameters passes the section
+  /// @param function is callable used to check crossing at the edges
+  template <typename F>
+  inline bool isLineInside(F function) const {
```

Comment:
If you think it's useful, you could add the concept requirement [`std::invocable`](https://en.cppreference.com/w/cpp/concepts/invocable.html) for better compiler error messages.